sqlserver/datadog_checks/sqlserver/activity.py

```diff
@@ -143,25 +194,77 @@ def _get_activity(self, cursor):
         rows = [dict(zip(columns, row)) for row in cursor.fetchall()]
         return rows
 
+    @tracked_method(agent_check_getter=agent_check_getter, track_result_length=True)
+    def _get_tx_activity(self, cursor):
+        self._time_since_last_tx_activity_event = time.time()
+        self.log.debug("collecting sql server transaction activity")
+        self.log.debug("Running query [%s]", TX_ACTIVITY_QUERY)
+        cursor.execute(TX_ACTIVITY_QUERY)
+        columns = [i[0] for i in cursor.description]
+        # construct row dicts manually as there's no DictCursor for pyodbc
+        rows = [dict(zip(columns, row)) for row in cursor.fetchall()]
+        self.check.histogram(
+            "dd.sqlserver.activity.get_tx_activity.tx_rows", len(rows), **self.check.debug_stats_kwargs()
+        )
+        self.log.warning(len(rows))
+        return rows
+
     def _normalize_queries_and_filter_rows(self, rows, max_bytes_limit):
         normalized_rows = []
+        row_key_and_index = {}
         estimated_size = 0
-        for row in rows:
-            try:
-                obfuscated_statement = obfuscate_sql_with_metadata(row['text'], self.check.obfuscator_options)['query']
-                row['query_signature'] = compute_sql_signature(obfuscated_statement)
-            except Exception as e:
-                # obfuscation errors are relatively common so only log them during debugging
-                self.log.debug("Failed to obfuscate query: %s", e)
-                obfuscated_statement = "ERROR: failed to obfuscate"
-            row = self._sanitize_row(row, obfuscated_statement)
+        for index, row in enumerate(rows):
+            row = self._obfuscate_and_sanitize_row(row)
             estimated_size += self._get_estimated_row_size_bytes(row)
             if estimated_size > max_bytes_limit:
-                # query results are ORDER BY transaction_begin_time ASC
+                # query results are ORDER BY query_start ASC
```

Comment:
As mentioned in the SQL, I believe the sort logic should live here. Otherwise there is no need to sort.